### PR TITLE
Allow running e2e test individually

### LINKED ISSUE #6228: Allow running e2e tests individually
In #6119, we added performance testing to this repo. To prevent that the performance tests run together with the e2e tests, we added the flags ` --testPathIgnorePatterns performance` to both the `test:e2e`, `test:e2e-dev` and `test:e2e-dev-watch` commands. Before we added these flags, we could run the entire test suite via `npm run test:e2e` and an individual test via `npm run test:e2e -- checkout-payments`. Since we added the flags, running an individual e2e test is no longer possible. 

This issue aims to ensure that we can run individual e2e tests again.

### PR DESCRIPTION
**Pull request by nielslange**
Fixes #6228 

### Notes

In #6228, we introduced performance testing. As a side effect, it was no longer possible to run an individual e2e test. This PR aims to fix this regression and allows running an individual e2e test again.

This fix follows a similar approach as Gutenberg, which is using the following two config files:
1. [jest.config.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/jest.config.js) for the e2e tests
2. [jest.performance.config.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/jest.performance.config.js) for the performance tests

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

### Manual Testing

1. Start Docker Desktop.
2. Run `npm run wp-env destroy && npm run wp-env start`.
3. Run `npm run test:performance` to ensure the performance tests are running.
4. Run `npm run test:e2e -- tax` to ensure an individual e2e test can be run.
5. Run `npm run test:e2e-dev -- tax` to ensure an individual e2e test can be run in dev mode.
6. Run `npm run test:e2e-dev-watch -- tax` to ensure an individual e2e test can be run in dev mode and watch for further adjustments.

### CODE REVIEW

File: tests/e2e/config/jest.performance.config.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+module.exports = {
```
Comment (alexflorisca) [inline tests/e2e/config/jest.performance.config.js:1]: At a glance, all of these settings are the same as `jest.config.js` so we could just require that here and extend the `reporters` property and add the peformance reporter to it. What do you think?

**Review by alexflorisca — COMMENTED** [review]
Woohoo, nice one @nielslange, this is great, it's been bugging me for ages. Just one comment about optimising `jest.performance.config.js`, otherwise looks good!

File: tests/e2e/config/jest.performance.config.js (hunk 1 above)
Comment (nielslange) [reply tests/e2e/config/jest.performance.config.js:1]: Great call, @alexflorisca! I've applied your suggestion directly.

**Review by alexflorisca — APPROVED** [review]
Looks good buddy, ship it 🚢

**Pull request merged**